src/v/raft/log_eviction_stm.cc

```diff
@@ -70,61 +219,131 @@ void log_eviction_stm::process_log_eviction() {
 }
 
 ss::future<> log_eviction_stm::housekeeping() {
+    bool should_sleep = true;
     try {
-        if (_eviction_offsets.empty()) {
-            co_await ss::sleep_abortable(200ms, _as);
-        } else {
-            auto eviction_offset = _eviction_offsets.front();
-            _requested_eviction_offset = eviction_offset;
-            if (eviction_offset > _raft->last_snapshot_index()) {
-                _eviction_offsets.pop_front();
-                co_await do_process_log_eviction(eviction_offset);
+        if (!_eviction_offsets.empty()) {
+            auto event = _eviction_offsets.front();
+            _eviction_offsets.pop_front();
+            _requested_eviction_offset = event.offset;
+            const auto log_start_offset = [this]() {
+                const auto ss_idx = _raft->last_snapshot_index();
```

Comment:
Shouldn't this be model::next_offset(_raft->last_snapshot_index())? `_raft->last_snapshot_index()` is the truncation point. no?

Reply:
Yeah but we need to take into consideration the newly added delta. The delta represents how far into the last_snapshotted_(segment) the new start offset is. This is only non-zero in the case a user made a delete-records request between a segment boundary

Reply:
Yes, I guess i was asking why you are avoiding the `+1` since you want the first visible offset here.